### PR TITLE
Port ammo-availability feature from HDR

### PR DESCRIPTION
**Pull request by CovenEsme**
## What does this PR do?
Adds the `ammo-availability` setting from HDR. Should work exactly the same. Also moves the post-LMF bird statue into the same place that HDR puts it to make it more visible to player used to the base-game.

## How do you test this changes?
* With the setting set to "Plentiful", there should be an ammo pot added next to every bird statue, next to the gossip stone in Thrill Digger, and before the crawlspace in Lanayru Caves from the Gorge entrance.
* When set to "Useful", ammo pots will be added in useful places to avoid back-tracking. This is basically the same as things are now before this PR
* When set to "Vanilla", no extra ammo pots are added
* When set to "Scarce", all guaranteed ammo drops are removed. Only chance-based drops may give ammo and ofc Rupin's Gear Shop

### CODE REVIEW

File: asm/custom-functions/src/game/flag_managers.rs [hunk 1]
```diff
@@ -131,3 +131,220 @@ impl DungeonflagManager {
         unsafe { (*Self::get_global(scn_idx))[1] & 0xF }
     }
 }
+
+// Flag enums
+#[repr(u16)]
+#[derive(Copy, Clone, Hash, PartialEq, Eq)]
+#[allow(non_camel_case_types)]
+pub enum ITEMFLAGS {
```
Comment (lepelog) [inline asm/custom-functions/src/game/flag_managers.rs:139]:
```suggestion
pub enum Itemflag {
```
I would follow the rust naming convention here and use singular

File: asm/custom-functions/src/rando/mod.rs [hunk 2]
```diff
@@ -530,3 +530,76 @@ pub fn has_heart_drops_enabled() -> c_int {
...
+    pos: *mut Vec3f,
+    _subtype: u32,
+    _rot: *mut c_void,
+) -> bool {
+    // 0xFE is the custom id being used to drop arrows, bombs, and seeds.
+    // Should set the eq flag for comparison after this addtion.
+    if param2_s0x18 == 0xFE {
+        extern "C" {
+            fn spawnDrop(itemid: ITEMFLAGS, roomid: u32, pos: *mut Vec3f, rot: *mut Vec3s);
+        }
```
Comment (lepelog) [inline asm/custom-functions/src/rando/mod.rs:548]: Imo this should go into `game/item.rs`

**Comment by CovenEsme** [comment]
Updated :p

**Comment by goldendevil1711** [comment]
What I did to test, all on Beta Features: https://nightly.link/ssrando/ssrando/workflows/build.yaml/beta-features/Skyward%20Sword%20Randomizer%20v2.2.0

Started a seed with all items, all dungeons open, on Plentiful mode. (String here: Z1zIyEMBAAAAAAAA4CCgAaIAPoFTAIA38/+P/QAAgP//AQAAAAAAAGAIAAAAAAAAAAAAAOAHAAAAAAAboP////////////8AAT4QPh0=#1240555879622308632)

Went to Sand Sea to break pots at Skipper's Retreat, Shipyard, and Pirate's Stronghold exterior Statues and the two interior Sandship statues, Caves on Gorge Side, and Gorge; as well as Faron at Lake Floria and Waterfall, and confirmed that they had ammo drops for arrows, bombs, and slingshot. I also checked Ancient Cistern. There was no ammo pot at the Ancient Cistern entrance statue, but there was a pot at the Koloktos door.

Started another seed with all items, all dungeons open, on Useful mode. (String here: Z1zIyEMBAAAAAAAA4CCgAaIAPoFTAIA38/+P/QAAgP//AQAAAAAAAGAIAAAAAAAAAAAAAOAHAAAAAAAboP////////////8AAT4QOBQ=#1240555879622308632)

Went to Lanayru to break pots at Mine Entry (no pots), Caves on Gorge side (pot with all 3 ammo), Gorge (no pot), West Desert (no pot), Desert South/Mine Entry (all 3 ammo), North Desert (no pot), and first LMF statue (all 3 ammo). Also went to Eldin to break pots at Entry Statue (no pots), Boko Base entry/East statue (bombs and arrows), Volcano East (no pots), Temple Entrance (no pots), ET first statue (all 3 ammo), ET center room (no pot)

Started another seed with all items, all dungeons open, on Scarce mode. (String here: Z1zIyEMBAAAAAAAA4CCgAaIAPoFTAIA38/+P/QAAgP//AQAAAAAAAGAIAAAAAAAAAAAAAOAHAAAAAAAboP////////////8AAT4QOAQ=#1240555879622308632)

Went to Lanayru to enter Sandship (lower deck and interior) and LMF (entry, Hub Room, Hop Across Room) and confirm that there were not ammo drops in the barrels in those areas or Fire Node entry. There were also no pots near statues at: Mine Entry, Caves (Gorge side), any of the Desert exterior statues

I did not have the time test Vanilla mode. I also did not chest, for example, end of LMF on Plentiful mode (which may be one specifically to check)

**Pull request merged**